.github/workflows/nightly.yml

```diff
@@ -91,3 +91,26 @@ jobs:
           path: wheelhouse/mlx_cuda-*.whl
           retention-days: 7
 
+  linux_fedora_build_cpp:
+    runs-on: ubuntu-22.04
```

Comment:
I would set this up as a single matrix build with something like `strategy: matrix: host: [ubuntu-22.04, ubuntu-22.04-arm]` and `runs-on: ${{ matrix.host }}` to visually reinforce that it's the same build but in two different environments. Syntax might need tweaking as I typed it from memory here.